src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -405,39 +410,64 @@ private static boolean isSummaryTag(DetailNode javadoc) {
     /**
      * Checks the inline summary (if present) for {@code period} at end and forbidden fragments.
      *
-     * @param ast javadoc root node.
+     * @param inlineSummaryTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
      */
-    private void validateSummaryTag(DetailNode ast) {
-        final String inlineSummary = getInlineSummary();
+    private void validateSummaryTag(DetailNode inlineSummaryTag) {
+        final String inlineSummary = getContentOfInlineCustomTag(inlineSummaryTag);
         final String summaryVisible = getVisibleContent(inlineSummary);
         if (summaryVisible.isEmpty()) {
-            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+            log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
         else if (!period.isEmpty()) {
             if (isPeriodAtEnd(summaryVisible, period)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+                log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
             }
             else if (containsForbiddenFragment(inlineSummary)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+                log(inlineSummaryTag.getLineNumber(), MSG_SUMMARY_JAVADOC);
             }
         }
     }
 
     /**
-     * Gets entire content of summary tag.
+     * Gets the content of inline custom tag.
      *
-     * @return summary sentence of javadoc root node.
+     * @param inlineTag inline tag node.
+     * @return String consisting of the content of inline custom tag.
      */
-    private String getInlineSummary() {
-        final DetailAST blockCommentAst = getBlockCommentAst();
-        final String javadocText = blockCommentAst.getFirstChild().getText();
-        final Matcher matcher = SUMMARY_PATTERN.matcher(javadocText);
-        String comment = "";
-        if (matcher.find()) {
-            comment = matcher.group(1);
+    public static String getContentOfInlineCustomTag(DetailNode inlineTag) {
+        final DetailNode[] childrenOfInlineTag = inlineTag.getChildren();
+        final StringBuilder customTagContent = new StringBuilder(256);
+        final int numOfChildrenEmptyInlineTag = 3;
+        if (childrenOfInlineTag.length != numOfChildrenEmptyInlineTag) {
+            final int indexOfContentOfSummaryTag = 3;
+            DetailNode currentNode = childrenOfInlineTag[indexOfContentOfSummaryTag];
```

Comment:
numOfChildrenEmptyInlineTag and indexOfContentOfSummaryTag are same value and  used to take value from same array.

Please use single variable.

Reply:
done